oauth-common/src/main/java/io/strimzi/kafka/oauth/common/OAuthAuthenticator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static io.strimzi.kafka.oauth.common.LogUtil.mask;
+import static io.strimzi.kafka.oauth.common.TokenIntrospection.introspectAccessToken;
+
+public class OAuthAuthenticator {
+
+    private static final Logger log = LoggerFactory.getLogger(OAuthAuthenticator.class);
+
+    public static TokenInfo loginWithAccessToken(String token) {
+        if (log.isDebugEnabled()) {
+            log.debug("loginWithAccessToken() - pass-through access_token: {}", mask(token));
+        }
+        // try introspect token
+        try {
+            return introspectAccessToken(token);
+        } catch (Exception e) {
+            log.debug("[IGNORED] Could not parse token as JWT access token. Could not extract scope, subject, and expiry.", e);
```

Comment:
maybe it's better a `log.warning` here?

Reply:
The reason I opted for debug logging here is that the token here can be an opaque token (not JWT) which will fail every time on this call. We could introduce another option to tell authenticator what kind of tokens we're dealing with. It's something I heaven's quite worked out how to deal with. The token will in any case be sent to the server, and rejected there if it's not valid.